sessions/2018/september/Session_12.md

```diff
@@ -0,0 +1,122 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 12 (18/09/2018)
+
+- Gradle(CLI tool for building java projects)
+- Generics (e.g. List and Map, and also custom ones)
+- Spring Initializr(website to quickly generate java (gradle) projects)
+
+### Notes:
+
+#### Gradle Installation
+
+following [Gradle installation guide][2].
+
+Firs I installed _SDKMAN_ according to instruction in [its website][1]:
+```bash
+$ curl -s "https://get.sdkman.io" | bash
+```
+So proceed to install Gradle:
+```bash
+$ sdk install gradle
+```
+
+In order to use it I followed [this tutorial][3] to make this example application:
+```java
+public static void main(String args[]){
+
+    LocalTime time = LocalTime.now();
+    System.out.println("Local time is = " + time);
+    System.out.println("Hello Gradle!");
+
+}
+```
+And this was my `build.grade` file:
+
+```java
+plugins {
+    //id 'java'
+    id 'application'
+}
+
+group 'co.org.osso'
+version '1.0-SNAPSHOT'
+
+mainClassName = 'demo.HelloGradle'
+
+sourceCompatibility = 1.8
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testCompile group: 'junit', name: 'junit', version: '4.12'
+
+}
+```
+Then, in order to compile and run with gradle the following commans must be done
+```bash
+[jaar@port-staff gradle-example]$ gradle build
+[jaar@port-staff gradle-example]$ gradle run
+```
+
+There is another step that is run with [Gradle Wrapper][4] in order to run projects without need an specific Gradle version (_to review_):
```

Comment:
Sorry, that's not clear at all. I was making this as a "comment" in order to point that this statement need to be reviewed, since I wasn't very sure what I was saying about. But I am thinking I could have done this in a github inline comment in my PR...

Reply:
Removing that stuff